### ISSUE TITLE
ActionSheetIOS does not work on iPad

### ISSUE THREAD

**Issue by coderdave** [facebook/react-native#3498]
ActionSheetIOS doesn't work in any of the iPad simulators or actual devices. No error, no console messages, just nothing...

Am I missing something here? Are you suppose to use something else for iPads? I don't see anything in the docs indicating otherwise.

**Comment by chirag04** [comment]
you need to provide more info/some code for us to help you.

Make sure you pass ``anchor`` in options for it to work on ipad.
Refer: https://github.com/facebook/react-native/blob/62e8ddc20561a39c3c839ab9f83c95493df117c0/Libraries/ActionSheetIOS/RCTActionSheetManager.m#L130-L139

I think anchor is not documented in the docs. cc @paramaggarwal

**Comment by coderdave** [comment]
Nothing out of the ordinary:
```javascript
  showActionSheet() {
    var BUTTONS = [
      'Option 1',
      'Option 2',
      'Cancel',
    ];
    ActionSheetIOS.showActionSheetWithOptions({
      options: BUTTONS,
      cancelButtonIndex: 2,
    },
    (buttonIndex) => {

      switch (buttonIndex) {
        case 0:
          console.log("Option 1");
          break
        case 1:
          console.log("Option 2");
          break
      }

    });
  },

  render() {
    return (
      <TouchableOpacity onPress={() => this.showActionSheet()}>
        <Text>Touch Me</Text>
      </TouchableOpacity>
    )
  }
```
In this case for example, what would I pass to `anchor`?

**Comment by chirag04** [comment]
I think this is a bug in ``showActionSheetWithOptions``. This bug was fixed in ``showShareActionSheetWithOptions`` by passing the ``anchor``.

``anchor`` is passed like this:
```anchor: React.findNodeHandle(this.refs.share), // Where you want the share popup to point to on iPad```

@paramaggarwal  want to give it a go?

**Comment by paramaggarwal** [comment]
@chirag04 Good point, I should document `anchor`, but this is not related to that. This bug does not affect `UIActionSheet` as far as I'm aware.

@coderdave from Apple's docs:

```
Important: UIActionSheet is deprecated in iOS 8. (Note that UIActionSheetDelegate is also deprecated.) To create and manage action sheets in iOS 8 and later, instead use UIAlertController with a preferredStyle of UIAlertControllerStyleActionSheet.
```
ref: https://developer.apple.com/library/prerelease/ios/documentation/UIKit/Reference/UIActionSheet_Class/index.html

**Comment by nicklockwood** [comment]
It's deprecated, but not removed. I suspect the issue is that action sheets need to be displayed inside a UIPopoverController on iPad, which is different from how they are used on iPhone. We probably don't have the correct code path implemented for iPad.

**Comment by coderdave** [comment]
If action sheets are deprecated and it's recommended to use `UIAlertControllerStyleActionSheet`, how would that be implemented in react native? Or has that not been done yet? Also, I guess I would have to target certain iOS versions to use the correct action sheet implementation.

**Comment by nicklockwood** [comment]
@coderdave it looks like I was wrong (not usual) - UIActionSheet does indeed seem to have stopped working when building with the iOS 9 SDK.

To implement UIAlertControllerStyleActionSheet we'll need to provide an alternative code path in the native module. For now, I think it's simplest if we stick with the same JS API and just change the native implementation.

As you guessed, we'll need to detect the OS version and provide different implementations, but fortunately that's quite easy to do in Objective-C. I'm working on it now.

**Comment by ide** [comment]
Should be fixed in a859d9e9315c2ff7
